api_app/api/routes/workspaces.py

```diff
@@ -543,3 +548,67 @@ async def retrieve_user_resource_operations_by_user_resource_id_and_operation_id
 async def retrieve_user_resource_history_by_user_resource_id(user_resource=Depends(get_user_resource_by_id_from_path), user=Depends(get_current_workspace_owner_or_researcher_user_or_airlock_manager), resource_history_repo=Depends(get_repository(ResourceHistoryRepository))) -> ResourceHistoryInList:
     validate_user_has_valid_role_for_user_resource(user, user_resource)
     return ResourceHistoryInList(resource_history=await resource_history_repo.get_resource_history_by_resource_id(resource_id=user_resource.id))
+
+@workspaces_shared_router.get("/workspaces/{workspace_id}/assignable-users", response_model=AssignableUsersInResponse, name=strings.API_GET_ASSIGNABLE_USERS)
```

Comment:
It might be tidier to move all the user routes into their own file? `users.py` maybe, and onto their own router?

Can the router be added conditionally based on that variable, rather than checking in each route? That way the endpoint doesn't even exist, which feels more secure.

Reply:
Will investigate the conditional piece, not sure.

Reply:
https://github.com/microsoft/AzureTRE/blob/a73cf2f32b62fce506b91b2b7f37ec2b2439de9f/api_app/api/routes/api.py#L36

have:

```
if USER_MANAGEMENT_ENABLED:
   core_router.include_router(users.users_router, tags=["users"])
```

or something like that.

Reply:
Have moved to `workspace_users.py` file and added the appropriate filter as recommended above.